src/datetime.ts

```diff
@@ -378,15 +333,20 @@ function lastOpts(argList) {
  * There's plenty others documented below. In addition, for more information on subtler topics like internationalization, time zones, alternative calendars, validity, and so on, see the external documentation.
  */
 export default class DateTime {
+  // Private readonly fields
+  private ts: Readonly<number>;
```

Comment:
Was there a reason to use the `Readonly` type over just making the attr readonly? 
`private readonly ts: number;`

Reply:
That was done for consistency. 

I understand primitives could support a `readonly` while Objects need a `Readonly<>` to have all their properties set to readonly.

I preferred to use the same keyword everywhere. Again, if this is not idiomatic, I'll be happy to fix.

Reply:
Sorry, but I think you misunderstood the meaning/usage of the [`Readonly<T>` Utility Type](https://www.typescriptlang.org/docs/handbook/utility-types.html#readonlyt) vs. the [`readonly` keyword](https://www.typescriptlang.org/docs/handbook/classes.html#readonly-modifier)...

The `readonly` **keyword** is used to make **a property** readonly (but not it's value/content).
```typescript
interface Bar { bazz: string }
class Foo {
    readonly n: number = 21
    readonly bar: Bar = { bazz: "foo" }
}
let foo = new Foo()
foo.n = 42 // <-- Error
foo.bar = { bazz: "bazz" } // <-- Error
foo.bar.bazz = "test" // <-- Working
```

The Readonly **utility type** (`Readonly<T>`) on the other hand modifies **all properties of the type T** to become readonly - e.g. to represent what [`Object.freeze`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) does. 
```typescript
function freeze<T>(obj: T): Readonly<T>;
```

All [Primitive values](https://developer.mozilla.org/en-US/docs/Glossary/Primitive) (like `number`, `string`, `boolean`, ...) already are **immutable** - they can't be altered. This means applying `Readonly<T>` to them (like  `Readonly<number>` or `Readonly<string>`) **has no effect at all!**

```typescript
interface Bar { bazz: string }
class Foo {
    n: number = 21
    i: Readonly<number> = 21  // makes no difference

    bar: Readonly<Bar> = { bazz: "foo" }  // basically the same as:
    bar2: { readonly bazz: string } = { bazz: "foo" }
}
let foo = new Foo()
foo.n = 42 // <-- Working
foo.i = 42 // <-- Working
foo.bar = { bazz: "bazz" } // <-- Working
foo.bar.bazz = "test" // <-- Error
foo.bar2 = { bazz: "bazz" } // <-- Working
foo.bar2.bazz = "test" // <-- Error
```

So if you really want the whole thing to be readonly you need: 
```typescript
interface Bar { bazz: string }
class Foo {
    readonly n: number = 21
    readonly bar: Readonly<Bar> = { bazz: "foo" }
}
let foo = new Foo()
foo.n = 42 // <-- Error
foo.i = 42 // <-- Error
foo.bar = { bazz: "bazz" } // <-- Error
foo.bar.bazz = "test" // <-- Error
```

Reply:
Thanks a lot, I thought I understood the difference between the two, and I chose to use `Readonly` everywhere for consistency. I did not realize it did not work as expected on primitives. Will fix.